sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/SolaceIO.java

```diff
@@ -1079,21 +1089,11 @@ public SolaceOutput expand(PCollection<T> input) {
                   MapElements.into(TypeDescriptor.of(Solace.Record.class))
                       .via(checkNotNull(getFormatFunction())));
 
-      // Store the current window used by the input
-      PCollection<Solace.PublishResult> captureWindow =
-          records.apply("Capture window", ParDo.of(new RecordToPublishResultDoFn()));
-
-      @SuppressWarnings("unchecked")
-      WindowingStrategy<Solace.PublishResult, BoundedWindow> windowingStrategy =
-          (WindowingStrategy<Solace.PublishResult, BoundedWindow>)
-              captureWindow.getWindowingStrategy();
-
       PCollection<Solace.Record> withGlobalWindow =
           records.apply("Global window", Window.into(new GlobalWindows()));
 
       PCollection<KV<Integer, Solace.Record>> withShardKeys =
-          withGlobalWindow.apply(
-              "Add shard key", ParDo.of(new AddShardKeyDoFn(getMaxNumOfUsedWorkers())));
+          withGlobalWindow.apply("Add shard key", ParDo.of(new AddShardKeyDoFn(getNumShards())));
```

Comment:
Agreed. We can add GroupIntoBatches in a separate pull request to avoid delaying this one. This ensures we can proceed without being held up by the GroupIntoBatches implementation, especially since it doesn't affect the public API.